apis/v1alpha2/gateway_types.go

```diff
@@ -557,8 +547,6 @@ type GatewayStatus struct {
 	// Listeners provide status for each unique listener port defined in the Spec.
 	//
 	// +optional
-	// +listType=map
-	// +listMapKey=port
```

Comment:
We should retain these annotations, but use the listener name as the `listMapKey`

Reply:
Added now - had to remove them when name was optional, but now that it's required it works well (and enforces uniqueness).